Source/Helpers/YPAdjustableView.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  YPAdjustableView.swift
+//
+//
+//  Created by Will Saults on 1/15/24.
+//
+
+import Photos
+import UIKit
+
+public class YPAdjustableView: UIView {
+    public var updateViewFrameAction: ((CGRect) -> Void)?
+
+    public func adjustViewFramesIfNeeded(cropRect: CGRect, asset: PHAsset, targetAspectRatio: CGFloat?) {
+        let assetSize = CGSize(width: CGFloat(asset.pixelWidth), height: CGFloat(asset.pixelHeight))
+
+        if let aspectRatio = targetAspectRatio, aspectRatio != assetSize.width / assetSize.height {
+            adjustPlayerViewFrameForAspectRatio(cropRect: cropRect, aspectRatio: aspectRatio, assetSize: assetSize)
+        } else {
+            adjustPlayerViewFrame(cropRect: cropRect, assetSize: assetSize)
+        }
+    }
+
+    // Method to adjust the player view frame considering the asset's original size
+    private func adjustPlayerViewFrame(cropRect: CGRect, assetSize: CGSize) {
+        let assetRect = CGRect(origin: .zero, size: assetSize)
+        let fitFrame = calculateAspectFitFrame(assetSize: assetSize)
+        let targetFrame = calculateTargetFrame(cropRect: cropRect, assetRect: assetRect, fitFrame: fitFrame)
+        updateViewFrameAction?(targetFrame)
+    }
+
+    // Method to adjust the player view frame based on a target aspect ratio
+    private func adjustPlayerViewFrameForAspectRatio(cropRect: CGRect, aspectRatio: CGFloat, assetSize: CGSize) {
```

Comment:
It looks like most of this code isn't new just migrated out of the extension?

Reply:
It's new for the video player. I moved it here to support the cover image. Looks like I need to change this naming.